### PR TITLE
CORE-20721: Add RBAC Group rest resource

### PR DESCRIPTION
**Pull request by Tom-Fitzpatrick**
This PR creates the GroupRestResource to allow creation and management of RBAC Groups. It includes all changes to support the Rest worker side of the work, up until sending the PermissionManagementRequest on Kafka. The DB side of this work will be continued in a following PR as this one is getting quite long. 

This is supported by api changes in https://github.com/corda/corda-runtime-os/pull/6231.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -39,7 +39,7 @@ commonsLangVersion = 3.12.0
 commonsTextVersion = 1.10.0
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.3.0.xx-SNAPSHOT to pick up maven local published copy
-cordaApiVersion=5.3.0.14-beta+
+cordaApiVersion=5.3.0.14-alpha-1719505397377
```
Comment (Tom-Fitzpatrick) [inline gradle.properties]: I will revert this once https://github.com/corda/corda-api/pull/1666 has been merged

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import net.corda.rest.annotations.HttpDELETE
+import net.corda.rest.annotations.HttpGET
+import net.corda.rest.annotations.HttpPOST
+import net.corda.rest.annotations.HttpPUT
+import net.corda.rest.annotations.RestPathParameter
+import net.corda.rest.response.ResponseEntity
+
+interface GroupEndpoint : RestResource {
+
+    @HttpPOST
```
Comment (vkolomeyko) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt]: Can we add `description`, `responseDescription` and `successCode` attributes to all the method annotations on this interface?

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import net.corda.rest.RestResource
+import net.corda.rest.annotations.ClientRequestBodyParameter
+import net.corda.rest.annotations.HttpDELETE
+import net.corda.rest.annotations.HttpGET
+import net.corda.rest.annotations.HttpPOST
+import net.corda.rest.annotations.HttpPUT
+import net.corda.rest.annotations.RestPathParameter
+import net.corda.rest.response.ResponseEntity
+
+interface GroupEndpoint : RestResource {
```
Comment (vkolomeyko) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt:26]: I think we do need `@HttpRestResource` annotation here. Please see `UserEndpoint` as a model.
Note that we need to set `minVersion = RestApiVersion.C5_3`.

Also, I would expect we would need to update `processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_3.json`.

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * The Group's role associations.
+     */
+    val roleAssociations: List<RoleAssociationResponseType>,
+
+    /**
+     * List of user IDs in the group.
+     */
+    val users: List<String>,
```
Comment (vkolomeyko) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt]:
```suggestion
    val users: Set<String>,
```

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * List of subgroups associated with the group.
+     */
+    val subgroups: List<String>
```
Comment (vkolomeyko) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt]:
```suggestion
    val subgroups: Set<String>
```

File: libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/response/GroupContentResponseDto.kt [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ */
+data class GroupContentResponseDto(
+    val id: String,
+    val lastUpdatedTimestamp: Instant,
+    val groupName: String,
+    val parentGroupId: String,
+    val properties: List<PropertyResponseDto>,
+    val roleAssociations: List<RoleAssociationResponseDto>,
+    val users: List<String>,
+    val subgroups: List<String>
```
Comment (vkolomeyko) [inline libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/response/GroupContentResponseDto.kt]:
```suggestion
    val users: Set<String>,
    val subgroups: Set<String>
```

File: libs/permissions/permission-manager-impl/src/test/kotlin/net/corda/libs/permissions/manager/impl/PermissionGroupManagerImplTest.kt [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import java.time.Duration
+import java.time.Instant
+import java.util.*
```
Comment (BenYip123) [inline libs/permissions/permission-manager-impl/src/test/kotlin/net/corda/libs/permissions/manager/impl/PermissionGroupManagerImplTest.kt]: Should remove this wildcard import

File: libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/request/DeleteGroupRequestDto.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package net.corda.libs.permissions.manager.request
+
+class DeleteGroupRequestDto(
+    val requestedBy: String,
```
Comment (BenYip123) [inline libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/request/DeleteGroupRequestDto.kt:7]: These will need a description aswell

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt (hunk 3 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt:26]: Ah thanks, have made these changes now.

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt (hunk 2 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt]: Done

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt (hunk 4 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt]: Accepted

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt (hunk 5 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/types/GroupContentResponseType.kt]: Accepted

File: libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/response/GroupContentResponseDto.kt (hunk 6 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/response/GroupContentResponseDto.kt]: Accepted

File: libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/request/DeleteGroupRequestDto.kt (hunk 8 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-manager/src/main/kotlin/net/corda/libs/permissions/manager/request/DeleteGroupRequestDto.kt:7]: Done, thank you!

File: libs/permissions/permission-manager-impl/src/test/kotlin/net/corda/libs/permissions/manager/impl/PermissionGroupManagerImplTest.kt (hunk 7 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-manager-impl/src/test/kotlin/net/corda/libs/permissions/manager/impl/PermissionGroupManagerImplTest.kt]: Done

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+            roleAssociations: A set of roles associated with the group"""
+    )
+    fun getGroupContent(
+        @RestPathParameter(description = "ID of the group to get content.")
+        groupId: String
+    ): GroupContentResponseType
+
+    @HttpDELETE(
+        path = "{groupId}",
+        description = "This method deletes a specified group.",
```
Comment (vkolomeyko) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/group/GroupEndpoint.kt]:
```suggestion
        description = "This method deletes a specified group. The group can only be deleted if it is empty, i.e. having no sub-groups or users inside it.",
```

**Review by vkolomeyko — COMMENTED** [review]
Looks fine, one suggestion posted.

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**